src/app/theme/components/baCheckbox/baCheckbox.component.ts

```diff
@@ -0,0 +1,31 @@
+import {Component, Provider, forwardRef, Input} from "@angular/core";
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from "@angular/common";
+
+const BA_CHECKBOX_CONTROL_VALUE_ACCESSOR = new Provider(
+  NG_VALUE_ACCESSOR, {
+    useExisting: forwardRef(() => BaCheckbox),
+    multi: true
+  });
+
+@Component({
+  selector: 'ba-checkbox',
+  template: require('./baCheckbox.html'),
+  providers: [BA_CHECKBOX_CONTROL_VALUE_ACCESSOR]
+})
+export class BaCheckbox implements ControlValueAccessor {
+  @Input() disabled:boolean = false;
+  @Input() label:string = 'Checkbox';
+  @Input() value:string = '';
+  @Input() name:string = '';
+  
+  private _state;
+
+  onChange(value: any): void {}
+  onTouch(value: any): void {}
+  writeValue(value: any): void {
+    this._state = value;
+  }
+
+  registerOnChange(fn: (_: any) => void): void { this.onChange = fn; }
```

Comment:
Shouldn't this be `registerOnChange(fn: (): any) => void): void { this.onChange = fn; }`